File: Programy z Materiałów (PDF'ów)/C++/Dzialania w systemach innych niz dziesietny/Dodawanie w systemie dwojkowym.cpp
#include <string>
#include <iostream>

using namespace std;

int main()
{
    int i, przn = 0, suma;
    string a,b,c="";
    cout << "Liczba 1: " << " ";
    cin >> a;
    cout << "Liczba 2: " << " ";
    cin >> b;
    while (a.size() < b.size()) a = '0' + a;
    while (b.size() < a.size()) b = '0' + b;
    for (i = a.size() - 1; i>=0; i--)
    {
        suma = przn + a[i] - '0' + b[i] - '0';
        if (suma % 2 == 1) c = '1' + c;
        else c = '0' + c;
        przn = suma / 2;
    }
    if (przn == 1) c = '1' + c;
    cout << "Suma: " << c << endl;
    return 0;
}

File: Programy z Materiałów (PDF'ów)/Python/Dzialania w systemach innych niz dziesietny/Dodawanie w systemie dwojkowym.py
a = input("Liczba 1: ")
b = input("Liczba 2: ")

while len(a) < len(b): a = '0' + a
while len(b) < len(a): b = '0' + b

c = ""

przn = 0

for i in range(len(a) - 1, -1, -1):
    suma = przn + int(a[i]) + int(b[i])
    if suma % 2 == 1: c = '1' + c
    else: c = '0' + c
    przn = suma // 2

if przn == 1: c = '1' + c

print("Suma: ", c)
